be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MeetingValidator.scala

```diff
@@ -70,24 +86,52 @@ sealed class MeetingValidator(dbActorRef: => AskableActorRef) extends MessageDat
         val laoId: Hash = rpcMessage.extractLaoId
         val expectedHash: Hash = Hash.fromStrings(EVENT_HASH_PREFIX, laoId.toString, data.creation.toString, data.name)
 
-        if (!validateTimestampStaleness(data.creation)) {
-          Right(validationError(s"stale 'creation' timestamp (${data.creation})"))
-        } else if (!validateTimestampOrder(data.creation, data.last_modified)) {
-          Right(validationError(s"'last_modified' (${data.last_modified}) timestamp is smaller than 'creation' (${data.creation})"))
-        } else if (!validateTimestampStaleness(data.start)) {
-          Right(validationError(s"stale 'start' timestamp (${data.start})"))
-        } else if (data.end.isDefined && !validateTimestampOrder(data.creation, data.end.get)) {
-          Right(validationError(s"'end' (${data.end.get}) timestamp is smaller than 'creation' (${data.creation})"))
-        } else if (data.end.isDefined && !validateTimestampOrder(data.start, data.end.get)) {
-          Right(validationError(s"'end' (${data.end.get}) timestamp is smaller than 'start' (${data.start})"))
-        } else if (!validateWitnessSignatures(data.modification_signatures, data.modification_id)) {
-          Right(validationError("witness key-signature pairs are not valid for the given modification_id"))
-        } else if (expectedHash != data.id) {
-          Right(validationError("unexpected id"))
-        } else {
-          Left(rpcMessage)
-        }
+        runChecks(
+          checkTimestampStaleness(
+            rpcMessage,
+            data.creation,
+            validationError(s"stale 'creation' timestamp (${data.creation})")
+          ),
+          checkTimestampOrder(
+            rpcMessage,
+            data.creation,
+            data.last_modified,
+            validationError(s"'last_modified' (${data.last_modified}) timestamp is smaller than 'creation' (${data.creation})")
+          ),
+          checkTimestampStaleness(
+            rpcMessage,
+            data.start,
+            validationError(s"stale 'start' timestamp (${data.start})")
+          ),
+          checkTimestampOrder(
+            rpcMessage,
+            data.creation,
+            data.end.get,
```

Comment:
Same as above, I think that line 109 could be problematic, if data.end is undefined an exception will be thrown instead of using the error pipeline.